core-bundle/src/Resources/contao/models/FilesModel.php

```diff
@@ -394,6 +395,51 @@ public static function findMultipleFoldersByFolder($strPath, array $arrOptions=a
 	protected function postSave($intType)
 	{
 	}
+
+	/**
+	 * Get the meta fields defined in `tl_files.meta.eval.metaFields`.
+	 */
+	public static function getMetaFields(): array
+	{
+		Controller::loadDataContainer('tl_files');
+
+		return array_keys($GLOBALS['TL_DCA']['tl_files']['fields']['meta']['eval']['metaFields'] ?? array());
+	}
+
+	/**
+	 * Get the meta data for this file. Specify one or more locales - meta
+	 * data of the first matching one will be returned or null if none was
+	 * found.
+	 */
+	public function getMetaData(string ...$locales): ?MetaData
+	{
+		$dataCollection = StringUtil::deserialize($this->meta, true);
+
+		foreach ($locales as $locale)
+		{
+			$data = $dataCollection[$locale] ?? array();
+
+			if (!empty($data))
+			{
```

Comment:
```suggestion
			if (empty($data))
			{
				continue;
			}
			
```
This would reduce the nesting level of the following statements, WDYT?

Reply:
Or even check directly for `empty($dataCollection[$locale])` two lines above?

Reply:
Simplified in c8c1f8f624bc18f7835ab78f665a8ade7b910fce.